plugins/azure-devops-backend/src/api/AzureDevOpsApi.ts

```diff
@@ -138,6 +142,103 @@ export class AzureDevOpsApi {
 
     return pullRequests;
   }
+
+  public async getBuildDefinitions(
+    projectName: string,
+    definitionName: string,
+  ): Promise<BuildDefinitionReference[]> {
+    this.logger?.debug(
+      `Calling Azure DevOps REST API, getting Build Definitions for ${definitionName} in Project ${projectName}`,
+    );
+
+    const client = await this.webApi.getBuildApi();
+    return client.getDefinitions(
+      projectName,
+      definitionName,
+      undefined,
+      undefined,
+      undefined,
+      undefined,
+      undefined,
+      undefined,
+      undefined,
+      undefined,
+      undefined,
+      undefined,
+      undefined,
+      undefined,
+      undefined,
+      undefined,
+      undefined,
+    );
+  }
+
+  public async getBuilds(
+    projectName: string,
+    top: number,
+    repoId?: string,
+    definitions?: number[],
+  ): Promise<Build[]> {
+    this.logger?.debug(
+      `Calling Azure DevOps REST API, getting up to ${top} Builds for Repository Id ${repoId} for Project ${projectName}`,
+    );
+
+    const client = await this.webApi.getBuildApi();
+    return client.getBuilds(
+      projectName,
+      definitions,
+      undefined,
+      undefined,
+      undefined,
+      undefined,
+      undefined,
+      undefined,
+      undefined,
+      undefined,
+      undefined,
+      undefined,
+      top,
+      undefined,
+      undefined,
+      undefined,
+      undefined,
+      undefined,
+      undefined,
+      repoId,
+      repoId ? 'TfsGit' : undefined,
+    );
+  }
+
+  public async getBuildRuns(
+    projectName: string,
+    top: number,
+    repoName?: string,
+    definitionName?: string,
+  ) {
+    let repoId: string | undefined;
+    let definitions: number[] | undefined;
+
+    if (repoName) {
+      const gitRepository = await this.getGitRepository(projectName, repoName);
+      repoId = gitRepository.id;
+    }
+
+    if (definitionName) {
+      const buildDefinitions = await this.getBuildDefinitions(
+        projectName,
+        definitionName,
+      );
+      definitions = buildDefinitions.map(bd => bd.id) as number[];
```

Comment:
I'm assuming this is to cast from `number | undefined[]` to `number[]`? You can do this so it ensures there's no undefined values:
```suggestion
      definitions = buildDefinitions.map(bd => bd.id).filter((bd ): bd is number => Boolean(bd ));
```

Reply:
This is cool, I've updated it to use this